src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -217,34 +238,56 @@ func (r *ClusterReconciler) createBootstrapConfigMap(
 	cluster *redpandav1alpha1.Cluster,
 	scheme *runtime.Scheme,
 ) error {
+	serviceAddress := cluster.Name + "." + cluster.Namespace + ".svc.cluster.local"
 	cfg := config.Default()
 	cfg.Redpanda = copyConfig(&cluster.Spec.Configuration, &cfg.Redpanda)
 	cfg.Redpanda.Id = 0
-	cfg.Redpanda.AdvertisedKafkaApi.Address = cluster.Name + seedSuffix + "-0" + "." +
-		cluster.Name + seedSuffix + "." +
-		cluster.Namespace +
-		".svc.cluster.local"
-	cfg.Redpanda.AdvertisedRPCAPI.Address = cluster.Name + seedSuffix + "-0" + "." +
-		cluster.Name + seedSuffix + "." +
-		cluster.Namespace +
-		".svc.cluster.local"
+	cfg.Redpanda.AdvertisedKafkaApi.Port = cfg.Redpanda.KafkaApi.Port
+	cfg.Redpanda.AdvertisedRPCAPI.Port = cfg.Redpanda.RPCServer.Port
 	cfg.Redpanda.Directory = dataDirectory
+	cfg.Redpanda.SeedServers = []config.SeedServer{
+		{
+			Host: config.SocketAddress{
+				// Example address: cluster-sample-0.cluster-sample.default.svc.cluster.local
+				Address:	cluster.Name + "-0." + serviceAddress,
+				Port:		cfg.Redpanda.AdvertisedRPCAPI.Port,
+			},
+			Id:	0,	// Start from 0 to match statefulset ordinals.
+		},
+	}
 
 	cfgBytes, err := yaml.Marshal(cfg)
 	if err != nil {
 		return err
 	}
 
+	script :=
+		`CONFIG=` + configPath + `;
+		ORDINAL_INDEX=${HOSTNAME##*-};
+		SERVICE_NAME=${HOSTNAME}.` + serviceAddress + `
+		cp /mnt/operator/redpanda.yaml $CONFIG;
+		rpk --config $CONFIG config set redpanda.node_id $ORDINAL_INDEX;
+		if [ "$ORDINAL_INDEX" = "0" ]; then
+			rpk --config $CONFIG config set redpanda.seed_servers '[]' --format yaml;
+		fi;
+		rpk --config $CONFIG config set redpanda.advertised_rpc_api.address $SERVICE_NAME;
+		rpk --config $CONFIG config set redpanda.advertised_rpc_api.port ` + strconv.Itoa(cfg.Redpanda.AdvertisedRPCAPI.Port) + `;
+		rpk --config $CONFIG config set redpanda.advertised_kafka_api.address $SERVICE_NAME;
+		rpk --config $CONFIG config set redpanda.advertised_kafka_api.port ` + strconv.Itoa(cfg.Redpanda.AdvertisedKafkaApi.Port) + `;
+		cat $CONFIG`
+
 	cm := &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace:	cluster.Namespace,
-			Name:		cluster.Name + seedSuffix,
+			Name:		cluster.Name + baseSuffix,
 			Labels:		cluster.Labels,
 		},
 		Data: map[string]string{
-			"redpanda.yaml": string(cfgBytes),
+			"redpanda.yaml":	string(cfgBytes),
```

Comment:
hmm I wonder, what are your thoughts around keeping these in one cfgmap vs two? I would personally split this but at the same time can't think of a strong reason for doing so... so just wanted to start a conversation

Reply:
maybe this is reason? https://github.com/vectorizedio/redpanda/pull/490/files#diff-208e6df471016ebc997b72c36741150b86b9023d2d8d698bbcb3a8e56f0f84d2R343

Reply:
The ConfigMap contains the script and the base configuration. Both of them are used during init only. So in that sense they bundle well together. On the other hand, I do see your point about chmod. My question would be, is it worth adding and managing one more ConfigMap?